content/post/dc-roma-ii-nix/index.md

```diff
@@ -0,0 +1,134 @@
+---
+title: Nix on the DC ROMA II
+description: Getting Nix bootstrapped on a RISC-V laptop using Ubuntu.
+date: 2025-11-17 11:50:00 +0800
+categories:
+    - riscv
+---
+
+Back in September, I was at NixCon 2025. I was able to meet Yuning Liang, the founder of DeepComputing.
+We had previously connected on LinkedIn several months prior. I had ordered their new laptop, the DC ROMA II.
+It has an ESWIN EIC7700X, an SoC with 8 out of order execution RV64GBC cores. This is a huge upgrade compared
+to the 4 in-order execution cores on the JH7110 that the VisionFive 2 uses. After coming back from a trip
+to Portland on the 8th, I started looking into getting Nix & NixOS working on it. Surely since I am able
+to port NixOS to things like the FydeTab Duo and get a VisionFive 2 to work, I should be able to get this system
+working.
+
+The first thing I did was install Nix, however there's no prebuilt binaries for Nix that the installer script
+utilizes. I took a look at the installer script and found a convinient option called `--tarball-url-prefix`.
+I spun up a build of Nix for RISC-V using an Arm cross environment. However, I ended up not needing that due
+to Hydra conviently having done that for me in
+[Build 312888635 of job nix:master:binaryTarballCross.x86_64-linux.riscv64-unknown-linux-gnu](https://hydra.nixos.org/build/312888635).
+I managed to even point `--tarball-url-prefix` to
+[`nix-2.33.0pre20251111_af0ac14-riscv64-linux.tar.xz`](https://hydra.nixos.org/build/312888635/download/1/nix-2.33.0pre20251111_af0ac14-riscv64-linux.tar.xz).
+Once I ran the installer, the installation experience was smooth. I restarted the shell and ran `nix --version` which told me that I have 2.33.0.
+
+After installing Nix, I decided to try building `nix-info`. I ran `nix-shell -p nix-info` and it was off to a smooth start.
+However, that soon changed as the `bootstrap-tools` failed to build. I realized this is a big problem, the `bootstrap-tools`
```

Comment:
Alternative phrasing for flow: "I ran `nix-shell -p nix-info` which seemed to start smoothly, but was quickly interrupted as `bootstrap-tools` failed to build. This is a big problem, as..."

Reply:
Done